src/nvim/eval.c

```diff
@@ -6007,6 +6012,26 @@ static void rettv_dict_alloc(typval_T *rettv)
   ++d->dv_refcount;
 }
 
+/// Clear the all the keys of a Dictionary.
+/// "d" remains a valid empty Dictionary.
+///
+/// @param d The Dictionary to clear
+void dict_clear(dict_T *d) {
```

Comment:
mutating functions like `dict_add` seem to assume non-NULL dict so I think this should do the same.